.github/actions/awaitStagingDeploys/awaitStagingDeploys.js

```diff
@@ -0,0 +1,33 @@
+const _ = require('underscore');
+const GitHubUtils = require('../../libs/GithubUtils');
+const {promiseDoWhile} = require('../../libs/promiseWhile');
+
+function run() {
+    let currentStagingDeploys = [];
+    return promiseDoWhile(
```

Comment:
FWIW our `async`/`await` ban really kills me here, in my opinion this would be _much_ cleaner written like so:

```js
const _ = require('underscore');
const GitHubUtils = require('../../libs/GithubUtils');

async function run() {
    let currentStagingDeploys = [];
    do {
        const response = await GitHubUtils.octokit.actions.listWorkflowRuns({
            owner: GitHubUtils.GITHUB_OWNER,
            repo: GitHubUtils.APP_REPO,
            workflow_id: 'platformDeploy.yml',
            event: 'push',
        });
        currentStagingDeploys = _.filter(response.data.workflow_runs, workflowRun => workflowRun.status !== 'completed');
        
        console.log(
            _.isEmpty(currentStagingDeploys)
                ? 'No current staging deploys found'
                : `Found ${currentStagingDeploys.length} staging deploy${currentStagingDeploys.length > 1 ? 's' : ''} still running...`,
        );
        
        await new Promise(resolve => setTimeout(resolve, GitHubUtils.POLL_RATE * 9))
    } while(!_.isEmpty(currentStagingDeploys));
}

if (require.main === module) {
    run();
}

module.exports = run;
```